src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1511,15 +1513,19 @@ def convert_to_parquet(fetched_data, fetch_idx):
         if fetch_size == 0:
             cursor.execute(query)
             result = cursor.fetchall()
-            parquet_file_queue.put(convert_to_parquet(result, 0))
+            parquet_file_path = convert_to_parquet(result, 0)
+            if parquet_file_path:
+                parquet_file_queue.put(parquet_file_path)
         elif fetch_size > 0:
             cursor = cursor.execute(query)
             fetch_idx = 0
             while True:
                 rows = cursor.fetchmany(fetch_size)
                 if not rows:
                     break
-                parquet_file_queue.put(convert_to_parquet(rows, fetch_idx))
+                parquet_file_path = convert_to_parquet(rows, fetch_idx)
+                if parquet_file_path:
+                    parquet_file_queue.put(parquet_file_path)
```

Comment:
we should use logger or other form to tell user that their data source is empty, otherwise user will only see they run dbapi and there is no data fetched

Reply:
we can do debug log

Reply:
added debug log in convert_to_parquet